core/utils/hashset/README.md

```diff
@@ -0,0 +1,25 @@
+# hashset.c
+
+A hash set implementation in C from:
+https://github.com/avsej/hashset.c
```

Comment:
Were fixes necessary in this code? If not, then why not include it as a submodule?

Reply:
The original hashset code had serious bugs, poor documentation, and poor generalizability. This is a pretty substantial rewrite. I suppose we could issue a PR in the original repo, but I don't think we want to depend on that repo.